src/main/java/graphql/nadel/engine/NadelExecutionStrategy.java

```diff
@@ -224,6 +232,47 @@ private Service getService(InnerServiceHydration innerServiceHydration) {
         return FpKit.findOne(services, service -> service.getName().equals(innerServiceHydration.getServiceName())).get();
     }
 
+    private CompletableFuture<ServiceExecutionResult> invokeService(Service service, ServiceExecution serviceExecution, ServiceExecutionParameters serviceExecutionParameters, ExecutionStepInfo executionStepInfo) {
+        try {
+            log.debug("service {} invocation started", service.getName());
+            CompletableFuture<ServiceExecutionResult> result = serviceExecution.execute(serviceExecutionParameters);
+            log.debug("service {} invocation finished ", service.getName());
+            //
+            // if they return an exceptional CF then we turn that into graphql errors as well
+            return result.handle(handleServiceException(service, executionStepInfo));
+        } catch (Exception e) {
+            return completedFuture(mkExceptionResult(service, executionStepInfo, e));
+        }
+    }
+
+    private BiFunction<ServiceExecutionResult, Throwable, ServiceExecutionResult> handleServiceException(Service service, ExecutionStepInfo executionStepInfo) {
+        return (serviceCallResult, throwable) -> {
+            if (throwable != null) {
+                return mkExceptionResult(service, executionStepInfo, throwable);
+            } else {
+                return serviceCallResult;
+            }
+        };
+    }
+
+    private ServiceExecutionResult mkExceptionResult(Service service, ExecutionStepInfo executionStepInfo, Throwable e) {
+        String errorText = String.format("An exception occurred invoking the service '%s' : '%s'", service.getName(), e.getMessage());
```

Comment:
Not sure if we want to show exception details to the client. Not necessarily in this PR but we should be able to define ExceptionMapper like we used to do in graphql-java that takes an exception and produces error message. It should also take care of logging so not error should be logged here.

Reply:
I think that can be done another PR